Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -35,39 +37,256 @@ struct DeclarationsSectionTranslator: RenderSectionTranslator {
+                                previousToken = .init(
+                                    text: previousToken.text,
+                                    kind: .text,
+                                    highlightDiff: nil)
+                            } else {
+                                // otherwise, split off the trailing whitespace into a new token and
+                                // save off the rest of the highlighted text
+                                let trimmedText = previousToken.text.removingTrailingWhitespace()
+                                let trailingWhitespace = previousToken.text.suffix(previousToken.text.count - trimmedText.count)
+                                processedDeclaration.append(.init(
+                                    text: trimmedText,
+                                    kind: .text,
+                                    highlightDiff: previousToken.highlightDiff))
 
-                    let declarationTokens = declarationFragments.map(translateFragment)
-                    otherDeclarations.append(
-                        .init(
-                            tokens: declarationTokens,
-                            identifier: overloadReference.absoluteString
-                        )
-                    )
+                                // save the trailing whitespace into `previousToken` so we can
+                                // potentially combine it with the next token below
+                                previousToken = .init(
+                                    text: String(trailingWhitespace),
+                                    kind: .text,
+                                    highlightDiff: nil)
+                            }
+                        } else if previousToken.highlightDiff != true,
+                                  token.highlightDiff == true,
+                                  token.kind == .text,
+                                  token.text.first?.isWhitespace == true
+                        {
+                            // if we're about to start a span of highlighted tokens with leading
+                            // whitespace, trim the space out of the highlighted section before
+                            // continuing
+                            if token.text.allSatisfy(\.isWhitespace) {
+                                // if this token is all whitespace, just convert it to be unhighlighted
+                                token = .init(
+                                    text: token.text,
+                                    kind: .text,
+                                    highlightDiff: nil)
+                            } else {
+                                // otherwise, split the whitespace into a new token and save the
+                                // remainder back into the token being processed
+                                let trimmedText = token.text.removingLeadingWhitespace()
```

Comment:
This particular logic is somewhat tied to the rendered declaration tokens, but i could see a generic version in the future if we expand this diffing logic to arbitrary content in the future. I don't think it's worth extracting this right now, though, since i'm not sure if that's something we want to do anyway.